test/test-functions

```diff
@@ -626,7 +626,8 @@ inst_libs() {
 }
 
 import_testdir() {
-    STATEFILE=".testdir"
+    STATEFILE="${BUILD_DIR:-.}/test/$(basename $(dirname $(realpath $0)))/.testdir"
+    mkdir -p $(dirname $STATEFILE)
```

Comment:
Indeed, fixed. While I was at it, I also used the same approach for `test.log`, so that `make -C test/TEST-01-BASIC` now works with a read-only tree too.